sandstone/pattern-video-player-custom/src/App/App.js

```diff
@@ -43,11 +45,25 @@ const AppBase = ({className, videoId, ...rest}) => {
 			hls.attachMedia(videoRef.current.getVideoNode().media);
 		} else {
 			hls.detachMedia();
-			const video = videoRef.current.getVideoNode().media;
-			video.src = source;
+			videoRef.current.getVideoNode().media.src = source;
 		}
 	}, [source, type]);
 
+	useEffect(() => {
+		const video = videoRef.current.getVideoNode().media;
+		if (video.textTracks[video.textTracks.length - 1]) {
+			video.textTracks[video.textTracks.length - 1].mode = "disabled";
+		}
+		if (subtitleVisible) {
+			const track = document.createElement('track');
+			track.src = subtitle;
+			track.kind = "subtitles";
+			track.default = true;
+			track.id = "track";
+			video.appendChild(track);
+		}
+	}, [subtitle, subtitleVisible]);
```

Comment:
Please check in the inspector that it has multiple <track> elements after toggling the sub button.
If it is true, I think it should be fixed.

As I'm not familiar with video track handling, You don't have to fix as same as my suggestion. 
```suggestion
	useEffect(() => {
		const video = videoRef.current.getVideoNode().media;
		let trackElement = document.getElementById('track');
		if (subtitle) {
			if (!document.getElementById('track')) {
				trackElement = document.createElement('track');
				trackElement.id = "track";
				video.appendChild(trackElement);
			}
			trackElement.src = subtitle;
			trackElement.kind = "subtitles";
		} else {
			video.textTracks[0].mode = "disabled";
		}
	}, [subtitle]);
```
Also, handleSubtitle need to be fixed together.
```
const handleSubtitle = useCallback(() => {
		const video = videoRef.current.getVideoNode().media;
		if (video.textTracks[0]) {
			video.textTracks[0].mode = subtitleVisible ? "disabled" : "showing"
		}
			setSubtitleVisible(!subtitleVisible)
		}, [subtitleVisible]
	);
```

Reply:
This code has a problem: previous subtitle remains after change the video.
So I changed code. Please check committed code.